docs/DeveloperGuide.md

```diff
@@ -1,35 +1,306 @@
+
+The Storage component can save the task list as TaskList objects in a .txt file format using Serialization and read it 
+back into a TaskList object.
+
+## Implementation
+
+> The lifeline of the sequence diagrams in this section should end at the destroy marker (X), but due to a limitation of PlantUML, the lifeline reaches the end of the diagram.
+
+### Delete task feature
+
+The DeleteTaskCommand extends NUS To-Do List with a delete feature for the removal of tasks from the task list.
+It is facilitated by ToDoListManager, Parser, exception, TaskList and Storage classes.
+It implements the `TaskList#deleteTask()` operation.
+
+Given below is an example usage scenario and how the DeleteTaskCommand mechanism behaves at each step.
+
+Step 1: The user launches the program for the first time. The ToDoListManager will be initialised. This in turn will
+then initialise the Parser, TaskList and Storage. Take it as there are no existing tasks read/stored by the program.
+
+Step 2: The user executes `add survey -due 20/03/2023 23:59` command to add a task for the To-Do List.
+The add command calls `TaskList#addTask()`, which causes a new Task to be added to the existing TaskList.
+
+Step 3: The user now then decides that adding this task was a mistake, and decides to remove the task from the
+To-Do List. The user does this by inputting the command `delete 1` into the terminal to delete a task in the task list.
+The command will then call the `TaskList#deleteTask()`, which removes the task at id 1 of the TaskList.
+
+The following sequence diagram shows how the delete task operation works:
+
+![DeleteTaskCommandSequence](images/DeleteTaskCommandSequence.png)
+
+Step 4: The user then decides to execute the command list. This command does not modify the TaskList.
+Thus, the TaskList will return to its initial state where there are no tasks stored in the TaskList.
+
+### Mark/unmark task feature
+
+Step 1: The user launches the program for the first time. The ToDoListManager will be initialised. This in turn will
+then initialise the Parser, TaskList and Storage. Take it as there are no existing tasks read/stored by the program.
+
+Step 2: The user executes `add survey -due 20/03/2023 23:59` command to add a task for the To-Do List.
+The add command calls `TaskList#addTask()`, which causes a new Task to be added to the existing TaskList.
+
+Step 3: The user wants to mark the task as completed by inputting the command `mark 1` into the terminal
+to mark the task as done. The command will then call the `TaskList#setDone`, which marks the task at id 1
+of the TaskList as done.
+
+For the unmark command, the user can instead input the command `unmark 1` to set the task as incomplete.
+The command also calls `TaskList#setDone` which sets the task at id 1 to be not done.
+
+The following sequence diagram shows how the mark/unmark task operation works:
+
+![MarkOrUnmarkTaskCommandSequence](images/MarkorUnmarkTaskCommandSequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/88194809/228707077-5c238e4e-c257-498a-a0a2-c96fc4605be7.png)
Should the return arrow be dotted line instead?